components/sup/src/manager/mod.rs

```diff
@@ -696,10 +705,15 @@ impl Manager {
                 self.shutdown(ShutdownReason::Departed);
                 return Err(sup_error!(Error::Departed));
             }
-            if let Some(SignalEvent::Passthrough(Signal::HUP)) = signals::check_for_signal() {
-                outputln!("Supervisor shutting down for signal");
-                self.shutdown(ShutdownReason::Signal);
-                return Ok(());
+            if env::var("HAB_SUP_IGNORE_SIGNAL").is_err() {
+                match signals::check_for_signal() {
+                    Some(SignalEvent::Passthrough(Signal::HUP)) | Some(SignalEvent::Shutdown) => {
+                        outputln!("Supervisor shutting down for signal");
+                        self.shutdown(ShutdownReason::Signal);
+                        return Ok(());
+                    }
+                    _ => {}
+                }
```

Comment:
Per @christophermaier's comment, if we're not also matching on `Some(SignalEvent::Shutdown)` (which we shouldn't), I think leaving this as the `if let` form is preferable to changing it to `match`.